src/Zicht/Itertools/lib/Traits/LastTrait.php

```diff
@@ -21,4 +21,19 @@ public function last($default = null)
     {
         return iter\last($this, $default);
     }
+
+    /**
+     * Returns the key of the last element of this iterable or
+     * returns $DEFAULT when this iterable is empty
+     *
+     * @param mixed $default
+     * @return mixed
+     */
+    public function lastKey($default = null)
+    {
+        $key = $default;
+        foreach ($this as $key => $value) {
+        }
+        return $key;
```

Comment:
Dat ik denk ik minder efficient, omdat dan de hele iterable in memory moet zitten om de reverse te doen.  Beide manieren moeten de hele lijst doorlopen, deze heeft als voordeel dat alleen de laatste in-memory blijft.